save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt

```diff
@@ -51,15 +60,28 @@ class TestExecutionService(private val testExecutionRepository: TestExecutionRep
      * @param testExecutionsDtos
      * @return list of lost tests
      */
+    @Suppress("TOO_MANY_LINES_IN_LAMBDA")
     fun saveTestResult(testExecutionsDtos: List<TestExecutionDto>): List<TestExecutionDto> {
         val lostTests: MutableList<TestExecutionDto> = mutableListOf()
+        val execution = agentRepository
+            .findByIdOrNull(testExecutionsDtos.first().agentId)  // we take agent id only from first element, because all test executions have same execution
+            ?.execution
+            ?: run {
+                log.error("Agent with id=[${testExecutionsDtos.first().agentId}] was not found in the DB")
+                return lostTests
+            }
         testExecutionsDtos.forEach { testExecDto ->
             val foundTestExec = testExecutionRepository.findById(testExecDto.id)
             foundTestExec.ifPresentOrElse({
                 it.run {
```

Comment:
I didn't think about it before, but it's a very strange statement here: we immediately swap `it` for `this` without any obvious improvements. Maybe, we should remove this `run`?